xdem/spatial_tools.py

```diff
@@ -398,6 +509,336 @@ def hillshade(dem: Union[np.ndarray, np.ma.masked_array], resolution: Union[floa
     # The output is scaled by "(x + 0.6) / 1.84" to make it more similar to GDAL.
     return np.clip(255 * (shaded + 0.6) / 1.84, 0, 255).astype("float32")
 
+def get_xy_rotated(raster: gu.georaster.Raster, along_track_angle: float) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Rotate x, y axes of image to get along- and cross-track distances.
+    :param raster: raster to get x,y positions from.
+    :param along_track_angle: angle by which to rotate axes (degrees)
+    :returns xxr, yyr: arrays corresponding to along (x) and cross (y) track distances.
+    """
+
+    myang = np.deg2rad(along_track_angle)
+
+    # get grid coordinates
+    xx, yy = raster.coords(grid=True)
+    xx -= np.min(xx)
+    yy -= np.min(yy)
+
+    # get rotated coordinates
+
+    # for along-track
+    xxr = np.multiply(xx, np.cos(myang)) + np.multiply(-1 * yy, np.sin(along_track_angle))
+    # for cross-track
+    yyr = np.multiply(xx, np.sin(myang)) + np.multiply(yy, np.cos(along_track_angle))
+
+    # re-initialize coordinate at zero
+    xxr -= np.nanmin(xxr)
+    yyr -= np.nanmin(yyr)
+
+    return xxr, yyr
+
+def rmse(z: np.ndarray) -> float:
+    """
+    Return root mean square error
+    :param z: Residuals between predicted and true value
+    :return: Root Mean Square Error
+    """
+    return np.sqrt(np.nanmean(np.square(np.asarray(z))))
+
+def huber_loss(z: np.ndarray) -> float:
+    """
+    Huber loss cost (reduces the weight of outliers)
+    :param z: Residuals between predicted and true values
+    :return: Huber cost
+    """
+    out = np.asarray(np.square(z) * 1.000)
+    out[np.where(z > 1)] = 2 * np.sqrt(z[np.where(z > 1)]) - 1
```

Comment:
Could it not be just one line:
`out = np.where(z > 1, 2 * np.sqrt(z[np.where(z > 1)]) - 1, np.square(z))`
Would be clearer, because it's hard to understand the metric as is written.

Reply:
Modified this way